### PR TITLE
feature/profile_API

### PR DESCRIPTION
**Pull request by ddanilyuk**

### CODE REVIEW

File: Owl/Modules/Common/ImagePicker.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+        return imagePicker
+    }
+
+    func updateUIViewController(
+        _ uiViewController: UIImagePickerController,
+        context: UIViewControllerRepresentableContext<ImagePicker>
+    ) {
+
+    }
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Common/ImagePicker.swift:32]: Empty `func`?

File: Owl/Modules/Common/ScrollViewDelegate.swift [hunk 2]
```diff
@@ -23,12 +23,10 @@ final class ScrollViewDelegate: NSObject, UIScrollViewDelegate, ObservableObject
     @Published var scrollViewDidEndDragging: ScrollViewChanged?
 
     func scrollViewDidScroll(_ scrollView: UIScrollView) {
-//        print("-----scrollViewDidScroll\(scrollView.contentOffset.y)")
         scrollViewDidScroll = ScrollViewChanged(scrollView: scrollView)
     }
 
     func scrollViewDidEndDragging(_ scrollView: UIScrollView, willDecelerate decelerate: Bool) {
-//        print("!!!!!scrollViewDidEndDragging\(scrollView.contentOffset.y)")
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Common/ScrollViewDelegate.swift:31]: 🔥

File: Owl/Modules/Login/EnterUserData/EnterUserDataStore.swift [hunk 3]
```diff
@@ -6,27 +6,39 @@
...
+
+        @BindableState var selectedImage: UIImage?
         @BindableState var firstName: String = ""
         @BindableState var lastName: String = ""
+
         var saveButtonEnabled: Bool = false
         var isLoading: Bool = false
 
         var alert: AlertState<Action>?
+        @BindableState var showImagePicker: Bool = false
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Login/EnterUserData/EnterUserDataStore.swift:25]: Move it to line `20`

File: Owl/Modules/Login/EnterUserData/EnterUserDataView.swift [hunk 4]
```diff
@@ -98,18 +99,24 @@ struct EnterUserDataView: View {
...
-                        dismiss: .dismissAlert
-                    )
                 }
             }
+            .disabled(viewStore.isLoading)
+            .overlay(
+                viewStore.isLoading
+                    ? Loader()
+                    : nil
+            )
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Login/EnterUserData/EnterUserDataView.swift]: Maybe inline?

File: Owl/Modules/Main/EditProfile/EditProfileStore.swift [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    struct State: Equatable {
+        @BindableState var selectedImage: UIImage?
+        @BindableState var firstName: String = ""
+        @BindableState var lastName: String = ""
+
+        var isLoading: Bool = false
+        var photo: Photo
+
+        var alert: AlertState<Action>?
+        @BindableState var showImagePicker: Bool = false
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Main/EditProfile/EditProfileStore.swift:24]: Move to line `19`

File: Owl/Modules/Main/EditProfile/EditProfileView.swift [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+    private enum Field: Int, CaseIterable {
+        case firstName
+        case lastName
+    }
+
+    var store: Store<EditProfile.State, EditProfile.Action>
+    @FocusState private var focusedField: Field?
+
+    var body: some View {
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Main/EditProfile/EditProfileView.swift:22]: Maybe better to divide body and provide `ZStack` and `VStack` as `var`s
It's to long

File: Owl/Modules/Main/EditProfile/EditProfileView.swift [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+                    .padding(20)
+                    .frame(minHeight: proxy.size.height)
+                }
+            }
+            .disabled(viewStore.isLoading)
+            .overlay(
+                viewStore.isLoading
+                    ? Loader()
+                    : nil
+            )
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Main/EditProfile/EditProfileView.swift]: Inline

**Review by AnastasiaHolovash — APPROVED** [review]

File: Owl/Modules/Common/ImagePicker.swift (hunk 1 above)
Comment (ddanilyuk) [reply Owl/Modules/Common/ImagePicker.swift:32]: Required by `UIViewControllerRepresentable`

**Pull request merged**